pkgs/development/python-modules/authlib/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, cryptography
+, requests
+}:
+
+buildPythonPackage rec {
+  version = "0.13";
+  pname = "authlib";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "Authlib";
+    sha256 = "0280wizg0q721dg95d55il5dlmmdw4y28c5gg4dmb9kr8cc4l6j0";
+  };
+
+  propagatedBuildInputs = [ cryptography requests ];
+
+  # no tests included
+  doCheck = false;
```

Comment:
since they have a github and do releases on it, do you mind checking out from github?

tests are really important in python to ensure that our disregard for versioning doesn't break anything

Reply:
@jonringer I failed getting the tests to work.

Look at it's `Makefile`, it uses `tox`. `tox` doesn't seem to discover python modules provided in `checkInputs`.
I tried going with `pytest` directly, but now ran into some relative import problems, and a lot of django warnings:

```
============================= test session starts ==============================
platform linux -- Python 3.7.5, pytest-5.2.1, py-1.8.0, pluggy-0.13.0
rootdir: /build/source, inifile: setup.cfg
collected 270 items / 16 errors / 254 selected                                 

==================================== ERRORS ====================================
______________ ERROR collecting tests/core/test_jose/test_jwe.py _______________
ImportError while importing test module '/build/source/tests/core/test_jose/test_jwe.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/core/test_jose/test_jwe.py:5: in <module>
    from tests.util import read_file_path
E   ModuleNotFoundError: No module named 'tests'
______________ ERROR collecting tests/core/test_jose/test_jwk.py _______________
ImportError while importing test module '/build/source/tests/core/test_jose/test_jwk.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/core/test_jose/test_jwk.py:5: in <module>
    from tests.util import read_file_path
E   ModuleNotFoundError: No module named 'tests'
______________ ERROR collecting tests/core/test_jose/test_jws.py _______________
ImportError while importing test module '/build/source/tests/core/test_jose/test_jws.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/core/test_jose/test_jws.py:5: in <module>
    from tests.util import read_file_path
E   ModuleNotFoundError: No module named 'tests'
______________ ERROR collecting tests/core/test_jose/test_jwt.py _______________
ImportError while importing test module '/build/source/tests/core/test_jose/test_jwt.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/core/test_jose/test_jwt.py:6: in <module>
    from tests.util import read_file_path
E   ModuleNotFoundError: No module named 'tests'
___ ERROR collecting tests/core/test_requests_client/test_oauth1_session.py ____
ImportError while importing test module '/build/source/tests/core/test_requests_client/test_oauth1_session.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/core/test_requests_client/test_oauth1_session.py:16: in <module>
    from tests.client_base import mock_text_response
E   ModuleNotFoundError: No module named 'tests'
___ ERROR collecting tests/core/test_requests_client/test_oauth2_session.py ____
ImportError while importing test module '/build/source/tests/core/test_requests_client/test_oauth2_session.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/core/test_requests_client/test_oauth2_session.py:13: in <module>
    from tests.util import read_file_path
E   ModuleNotFoundError: No module named 'tests'
_________ ERROR collecting tests/django/test_oauth1/test_authorize.py __________
tests/django/test_oauth1/test_authorize.py:4: in <module>
    from .models import User, Client
tests/django/test_oauth1/models.py:3: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
_____ ERROR collecting tests/django/test_oauth1/test_resource_protector.py _____
tests/django/test_oauth1/test_resource_protector.py:10: in <module>
    from .models import User, Client, TokenCredential
tests/django/test_oauth1/models.py:3: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
_____ ERROR collecting tests/django/test_oauth1/test_token_credentials.py ______
tests/django/test_oauth1/test_token_credentials.py:6: in <module>
    from .models import User, Client
tests/django/test_oauth1/models.py:3: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
__ ERROR collecting tests/django/test_oauth2/test_authorization_code_grant.py __
tests/django/test_oauth2/test_authorization_code_grant.py:5: in <module>
    from .models import User, Client
tests/django/test_oauth2/models.py:10: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
__ ERROR collecting tests/django/test_oauth2/test_client_credentials_grant.py __
tests/django/test_oauth2/test_client_credentials_grant.py:3: in <module>
    from .oauth2_server import TestCase
tests/django/test_oauth2/oauth2_server.py:5: in <module>
    from .models import Client, OAuth2Token
tests/django/test_oauth2/models.py:10: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
_______ ERROR collecting tests/django/test_oauth2/test_implicit_grant.py _______
tests/django/test_oauth2/test_implicit_grant.py:3: in <module>
    from .oauth2_server import TestCase
tests/django/test_oauth2/oauth2_server.py:5: in <module>
    from .models import Client, OAuth2Token
tests/django/test_oauth2/models.py:10: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
_______ ERROR collecting tests/django/test_oauth2/test_password_grant.py _______
tests/django/test_oauth2/test_password_grant.py:6: in <module>
    from .oauth2_server import TestCase
tests/django/test_oauth2/oauth2_server.py:5: in <module>
    from .models import Client, OAuth2Token
tests/django/test_oauth2/models.py:10: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
_______ ERROR collecting tests/django/test_oauth2/test_refresh_token.py ________
tests/django/test_oauth2/test_refresh_token.py:5: in <module>
    from .models import User, Client, OAuth2Token
tests/django/test_oauth2/models.py:10: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
_____ ERROR collecting tests/django/test_oauth2/test_resource_protector.py _____
tests/django/test_oauth2/test_resource_protector.py:4: in <module>
    from .models import User, Client, OAuth2Token
tests/django/test_oauth2/models.py:10: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
____ ERROR collecting tests/django/test_oauth2/test_revocation_endpoint.py _____
tests/django/test_oauth2/test_revocation_endpoint.py:3: in <module>
    from .oauth2_server import TestCase
tests/django/test_oauth2/oauth2_server.py:5: in <module>
    from .models import Client, OAuth2Token
tests/django/test_oauth2/models.py:10: in <module>
    from django.contrib.auth.models import User
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/models.py:4: in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:52: in <module>
    class AbstractBaseUser(models.Model):
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/contrib/auth/base_user.py:53: in AbstractBaseUser
    password = models.CharField(_('password'), max_length=128)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1061: in __init__
    super(CharField, self).__init__(*args, **kwargs)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/fields/__init__.py:172: in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:56: in __getattr__
    self._setup(name)
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/conf/__init__.py:39: in _setup
    % (desc, ENVIRONMENT_VARIABLE))
E   django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
=============================== warnings summary ===============================
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/sql/query.py:11
/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/sql/query.py:11
  /nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/db/models/sql/query.py:11: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    from collections import Counter, Iterator, Mapping, OrderedDict

/nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/core/paginator.py:126
  /nix/store/zn23ai5lyswfml3clma07zq7lkpw89f0-python3.7-Django-1.11.24/lib/python3.7/site-packages/django/core/paginator.py:126: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    class Page(collections.Sequence):

-- Docs: https://docs.pytest.org/en/latest/warnings.html
!!!!!!!!!!!!!!!!!!! Interrupted: 16 errors during collection !!!!!!!!!!!!!!!!!!!
======================== 3 warnings, 16 error in 1.98s =========================
builder for '/nix/store/j4vwxsnz8ga86qhcdj3bp3n7qm6fbp35-python3.7-authlib-0.13.drv' failed with exit code 2
error: build of '/nix/store/j4vwxsnz8ga86qhcdj3bp3n7qm6fbp35-python3.7-authlib-0.13.drv' failed
```

That's my diff so far:

```patch
diff --git a/pkgs/development/python-modules/authlib/default.nix b/pkgs/development/python-modules/authlib/default.nix
index 1f79bd3da8f..5cc8a9becb1 100644
--- a/pkgs/development/python-modules/authlib/default.nix
+++ b/pkgs/development/python-modules/authlib/default.nix
@@ -1,24 +1,31 @@
 { stdenv
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , cryptography
+, flask
 , requests
+, python3
 }:
 
 buildPythonPackage rec {
   version = "0.13";
   pname = "authlib";
 
-  src = fetchPypi {
-    inherit version;
-    pname = "Authlib";
-    sha256 = "0280wizg0q721dg95d55il5dlmmdw4y28c5gg4dmb9kr8cc4l6j0";
+  src = fetchFromGitHub {
+    owner = "lepture";
+    repo = "authlib";
+    rev = "v${version}";
+    sha256 = "1nv0jbsaqr9qjn7nnl55s42iyx655k7fsj8hs69652lqnfn5y3d5";
   };
 
-  propagatedBuildInputs = [ cryptography requests ];
+  propagatedBuildInputs = [ cryptography ];
 
-  # no tests included
-  doCheck = false;
+  # FIXME import individually
+  checkInputs = with python3.pkgs; [ django flask flask_sqlalchemy tox mock pytest requests coverage ];
+
+  checkPhase = ''
+    py.test . --ignore tests/py3/test_httpx_client --ignore tests/py3/test_starlette_client
+  '';
```

If you know how to make this work, I'd appreciate if you could take over.

Reply:
the flask, django, and py3 directories seem to be more integration oriented. Here's how I would write this expression:

```
{ stdenv
, buildPythonPackage
, fetchFromGitHub
, pytest
, mock
, cryptography
, requests
}:

buildPythonPackage rec {
  version = "0.13";
  pname = "authlib";

  src = fetchFromGitHub {
    owner = "lepture";
    repo = pname;
    rev = "v${version}";
    sha256 = "1nv0jbsaqr9qjn7nnl55s42iyx655k7fsj8hs69652lqnfn5y3d5";
  };

  propagatedBuildInputs = [ cryptography requests ];

  # run unit tests
  checkInputs = [ pytest mock ];
  # source root needs to be on pythonpath due to test imports
  checkPhase = ''
    PYTHONPATH=$PWD:$PYTHONPATH pytest tests/{core,files}
  '';

  meta = with stdenv.lib; {
    homepage = https://github.com/lepture/authlib;
    description = "The ultimate Python library in building OAuth and OpenID Connect servers. JWS,JWE,JWK,JWA,JWT included.";
    maintainers = with maintainers; [ flokli ];
    license = licenses.bsd3;
  };
}```

Reply:
right, `PYTHONPATH=$PWD` was what I was missing :man_facepalming: 

Thanks! Updated the PR with your suggestions.